### DISCUSSION TITLE
Sequelize - Order By before Include

### DISCUSSION THREAD

**Discussion by PedroCostaMudey** [sequelize/sequelize#14124] [Help & Questions]
I have a main table where I join other tables based in an id (is not a FK nor PK).

I use include with this other tables and everything works fine.

Problem is that I want to order the main table by create_at before doing any includes.

I can do this easy with mysql I just do a sub query in the main table and order by create_at and then I join the other tables. 

I need this because the other tables have a huge amount of data in some fields and so db runs out of sorting memory.

**Comment by PedroCostaMudey** [comment]
Can you do something like attribute but instead of existing after SELECT it could exist after FROM so I could use a squelize.literal() to do a subquery there?

**Reply by ephys** [reply]
I don't know if it is going to work but `findAll` has a `subQuery` option you can turn on

**Reply by PedroCostaMudey** [reply]
The subQuery seams to create subQuery in more than one place. 

Also creates some strange Errors when used with findAndCountAll()
I have segregated it into findAll() and count() and it works but still have subqueries in places where they will create the same out of sorting memory in the mysql db because its trying to apply order by in all of the included models. 

I'm doing multiple joins and really only want a subQuery in the main model not in the other models that are included.

**Comment by wirujs** [comment]
Hi, does anyone have any update on this?
